stm/stm.ml

```diff
@@ -2234,10 +2234,13 @@ let known_state ?(redefine_qed=false) ~cache id =
 	    if eff then update_global_env ()
           ), (if eff then `Yes else cache), true
       | `Cmd { cast = x; ceff = eff } -> (fun () ->
-            reach view.next;
-            stm_vernac_interp id x;
-	    if eff then update_global_env ()
-          ), (if eff then `Yes else cache), true
+          (match !Flags.async_proofs_mode with
```

Comment:
Fair points. I should probably just rely on `BuildVio`, so that errors are ignored except when compiling with `-quick`, but I'd like to do some testing first.

I appreciate the sentiment but I don't think I understand what's going on - hopefully I understand enough at some point to feel confident this is a real fix.